<h1>
 Title: 2016 fall 7:30 prelim 5
</h1>
<h3>
 Author: Anonymous
</h3>
<h3>
 Date: 2022-03-07T15:33:55.444361+11:00
</h3>
<h3>
 Category: Exams
</h3>
<h3>
 Vote Count: 0
</h3>
<p>
 I'm having trouble with exception handling problems. Could someone explain the solution to C.me(0) and C.me(1)?
</p>
<figure>
 <image height="632.3636363636364" src="https://static.us.edusercontent.com/files/c4uRESQlRehJFaz1EHuzVGna" width="658"/>
</figure>
<h3>
 ----------- REPLIES -----------
</h3>
<div style="text-indent: 2em;">
 <h3>
  Author: Vladia Trinh (staff)
 </h3>
 <h3>
  Vote Count: 0
 </h3>
 <p>
  For C.m(0)...
 </p>
 <list style="bullet">
  <li>
   <p>
    print 8
   </p>
  </li>
  <li>
   <p>
    y is assigned to 0
   </p>
  </li>
  <li>
   <p>
    print 7
   </p>
  </li>
  <li>
   <p>
    if statement evaluates to true, so runtime exception is thrown
   </p>
  </li>
  <li>
   <p>
    runtime exception is caught in catch branch so print 2
   </p>
  </li>
  <li>
   <p>
    lastly print 1
   </p>
  </li>
 </list>
 <p>
  So, you end up printing 8 7 2 1
 </p>
 <p>
  For C.m(1)...
 </p>
 <list style="bullet">
  <li>
   <p>
    print 8
   </p>
  </li>
  <li>
   <p>
    evaluating expression to assign y divides by 0, since this is not in the try statement it cannot be caught
   </p>
  </li>
  <li>
   <p>
    program crashes
   </p>
  </li>
 </list>
 <p>
  So, you end up just printing 8
 </p>
 <h3>
  ------------------------------------
 </h3>
</div>